### ISSUE TITLE
The documentation is out of date

### ISSUE THREAD

**Issue by akviacode** [microsoft/windows-admin-center-sdk#59]
Hello,

We've updated @microsoft/windows-admin-center-sdk
Now we cannot build our solutions: there are lot of changes in sdk. And it is not documented anywhere.
E.g.
https://docs.microsoft.com/en-us/windows-server/manage/windows-admin-center/extend/guides/create-connection-provider
Here there are code samples, but the new sdk has no this.appContextService.rpc.updateData method anymore. So how can we add new connections?
Paths for imported modules were changed too.
Can you provide uptodate documentation for developers?

Also, Developer Tools extension is not available now. Are you bring it back and when?

Thanks

**Comment by mattatmsft** [comment]
Hi there - 

Which version of the SDK were you on previously?  And did you update to latest or insider?

**Comment by akviacode** [comment]
Updated to the latest:
`"@microsoft/windows-admin-center-sdk": "latest"`

Previous version was
`"version": "0.856.0"`

Now it is:
`"version": "1.278.0"`

Also, we've noticed that angular version was changed from 5.1.2 to 7.1.1

**Comment by mattatmsft** [comment]
Yes, that was a really big change.  A very fundamental platform update, which did cause a lot of update work.

My suggestion would be to use [the CLI](https://www.npmjs.com/package/windows-admin-center-cli) to create a new project and them move your functional components from the old project into the new.

Also check out [this document](https://docs.microsoft.com/en-us/windows-server/manage/windows-admin-center/extend/guides/migration-guide-0_1-1_0) on migrating.

**Comment by akviacode** [comment]
Great document! Thanks. Also msft.sme.shell.devguide is awesome.

 Have read all. Last point here, previous document **Connection Provider example** code sample was:

```
public onSubmit() {
let connections: Connection[] = [];

let connection = <Connection> {
  id: ConnectionUtility.createConnectionId(this.connectionType, this.newConnectionName),
  type: this.connectionType,
  name: this.newConnectionName
};

connections.push(connection);

this.appContextService.rpc.updateData(
  EnvironmentModule.nameOfShell,
  '##',
  <RpcUpdateData> {
	results: {
	  connections: connections,
	  credentials: null
	}
  }
);
}

public onCancel() {
this.appContextService.rpc.updateData(
  EnvironmentModule.nameOfShell, '##', <RpcUpdateData>{ results: { connections: [] } });
}
```

It should be replaced with:

```
public onSubmit() {
    const connection = <Connection> {
        id: ConnectionUtility.createConnectionId(this.connectionType, this.newConnectionName),
        type: this.connectionType,
        name: this.newConnectionName
      };

      this.appContextService.connectionManager.addOrUpdateConnection(connection);

      // TODO how to close dialog?
  }

  public onCancel() {
   // TODO how to close dialog?
  }
```

Is it correct? If so, how this dialog can be closed?

It's the last migration problem, all other were migrated perfect.

**Comment by Harrioson** [comment]
Microsoft window is an operating system, 
which is needed to be loaded in a computer after purchasing it.
https://www.techsovibe.com/microsoft-windows/

**Comment by donatwork** [comment]
> Yes, that was a really big change. A very fundamental platform update, which did cause a lot of update work.
> 
> My suggestion would be to use [the CLI](https://www.npmjs.com/package/windows-admin-center-cli) to create a new project and them move your functional components from the old project into the new.
> 
> Also check out [this document](https://docs.microsoft.com/en-us/windows-server/manage/windows-admin-center/extend/guides/migration-guide-0_1-1_0) on migrating.

That link no longer works, getting 404 error. I do remember seeing that link a year ago but back then I was not working on the project.

**Comment by donatwork** [comment]
>       // TODO how to close dialog?

Out of curiosity, how did you close the dialog(s)?

**Comment by Saur1bh** [comment]
<img width="688" alt="image" src="https://github.com/microsoft/windows-admin-center-sdk/assets/152589975/c80fe781-854f-49dc-8d18-868abf57a171">
 While implementing below code getting this error, Can anyone help on this

public onSubmit() {
    const connection = <Connection> {
        id: ConnectionUtility.createConnectionId(this.connectionType, this.newConnectionName),
        type: this.connectionType,
        name: this.newConnectionName
      };

      this.appContextService.connectionManager.addOrUpdateConnection(connection);

  }
